app/Helpers/database/player-game.php

```diff
@@ -276,20 +277,33 @@ function reactivateUserEventAchievements(User $user, array $userUnlocks): array
 
 function GetAllUserProgress(User $user, int $consoleID): array
 {
-    $retVal = [];
-
-    $query = "SELECT gd.ID, gd.achievements_published AS NumAch,
-                     COALESCE(pg.achievements_unlocked, 0) AS Earned,
-                     COALESCE(pg.achievements_unlocked_hardcore, 0) AS HCEarned
-            FROM GameData AS gd
-            LEFT JOIN player_games pg ON pg.game_id = gd.ID AND pg.user_id={$user->id}
-            WHERE gd.achievements_published > 0 AND gd.ConsoleID = $consoleID";
+    /** @var Collection<int, PlayerGame> $playerGames */
+    $playerGames = $user->playerGames()
+        ->whereIn('game_id', function ($query) use ($consoleID) {
+            $query->select('ID')
+                  ->from('GameData')
+                  ->where('ConsoleID', $consoleID)
+                  ->where('achievements_published', '>', 0);
+        })
+        ->get()
+        ->keyBy('game_id');
 
-    foreach (legacyDbFetchAll($query) as $row) {
-        $id = $row['ID'];
-        unset($row['ID']);
+    /** @var Collection<int, Game> $games */
+    $games = Game::where('ConsoleID', $consoleID)
+        ->where('achievements_published', '>', 0)
+        ->get();
```

Comment:
Would it make sense to reuse this query for the filter above? (psuedo-code incoming)
```
$playerGames = $user->playerGames()
    ->whereIn('game_id', $games->pluck('ID'))
```

Reply:
It would -- in latest I've actually condensed this down to a single query.